internal/pkg/cache/cache_test.go

```diff
@@ -403,3 +525,67 @@ func TestParsePackURL(t *testing.T) {
 		})
 	}
 }
+
+type TestLogger struct {
+	t *testing.T
+}
+
+// Debug logs at the DEBUG log level
+func (l *TestLogger) Debug(message string) {
+	l.t.Helper()
+	l.t.Log(message)
+}
+
+// Error logs at the ERROR log level
+func (l *TestLogger) Error(message string) {
+	l.t.Helper()
+	l.t.Log(message)
+}
+
+// ErrorWithContext logs at the ERROR log level including additional context so
+// users can easily identify issues.
+func (l *TestLogger) ErrorWithContext(err error, sub string, ctx ...string) {
+	l.t.Helper()
+	l.t.Log(fmt.Sprintf("err: %s", err))
+	l.t.Log(sub)
+	for _, entry := range ctx {
+		l.t.Log(entry)
+	}
+}
+
+// Info logs at the INFO log level
+func (l *TestLogger) Info(message string) {
+	l.t.Helper()
+	l.t.Log(message)
+}
+
+// Trace logs at the TRACE log level
+func (l *TestLogger) Trace(message string) {
+	l.t.Helper()
+	l.t.Log(message)
+}
+
+// Warning logs at the WARN log level
+func (l *TestLogger) Warning(message string) {
+	l.t.Helper()
+	l.t.Log(message)
+}
+
+// NewTestLogger returns a test logger suitable for use with the go testing.T log function.
+func NewTestLogger(t *testing.T) *TestLogger {
+	return &TestLogger{
+		t: t,
+	}
+}
+
+// NoopLogger returns a logger that meets the Logger interface, but does nothing
+// for any of the methods. This can be useful for cases that require a Logger as
+// a parameter.
+type NoopLogger struct{}
+
+func (_ NoopLogger) Debug(_ string)                                  {}
+func (_ NoopLogger) Error(_ string)                                  {}
+func (_ NoopLogger) ErrorWithContext(_ error, _ string, _ ...string) {}
+func (_ NoopLogger) Info(_ string)                                   {}
+func (_ NoopLogger) Trace(_ string)                                  {}
+func (_ NoopLogger) Warning(_ string)                                {}
```

Comment:
Does this need to be defined here as well in addition to `internal/pkg/logging/logger.go`?